release-note-script/src/main/java/ReleaseNoteCreation.java

```diff
@@ -0,0 +1,331 @@
+    }
+  }
+
+  static class ReleaseNoteText {
+    public Category category;
+    public String text;
+    public List<String> prNumbers = new ArrayList<>();
+  }
+
+  public static class GitHubContext {
+
+    private static final String MERGED_STATE = "merged";
+
+    private final String owner;
+    private final String projectTitlePrefix;
+    private final String version;
+    private final String repository;
+
+    public GitHubContext(String owner, String projectTitString, String version, String repository) {
+      this.owner = owner;
+      this.projectTitlePrefix = projectTitString;
+      this.version = version;
+      this.repository = repository;
+    }
+
+    public String getProjectId() throws Exception {
+      BufferedReader br =
+          runSubProcessAndGetOutputAsReader(
+              format(
+                  "gh project list --owner %s | awk '/%s/ {print}' | awk '/%s/ {print $1}'",
+                  this.owner, this.projectTitlePrefix, this.version));
+
+      String line = br.readLine(); // Assuming only one line exists.
+      if (line == null) throw new RuntimeException("Couldn't get the projectId");
+      return line;
+    }
+
+    public List<String> getPullRequestNumbers(String projectId) throws Exception {
+      BufferedReader br =
+          runSubProcessAndGetOutputAsReader(
+              format(
+                  "gh project item-list %s --owner %s --limit 200 | awk -F'\\t' '/%s\\t/ {print"
+                      + " $3}'",
+                  projectId, this.owner, this.repository));
+
+      String line;
+      List<String> prNumbers = new ArrayList<>();
+      while ((line = br.readLine()) != null) {
+        prNumbers.add(line);
+      }
+      return prNumbers;
+    }
+
+    private String getPullRequestState(String prNumber) throws Exception {
+      BufferedReader br =
+          runSubProcessAndGetOutputAsReader(
+              format(
+                  "gh pr view %s --repo %s/%s --jq \".state\" --json state",
+                  prNumber, this.owner, this.repository));
+
+      String line = br.readLine(); // Assuming only one line exists.
+      if (line == null) throw new RuntimeException("Couldn't get the projectId");
+      return line;
+    }
+
+    public boolean isPullRequestMerged(String prNumber) throws Exception {
+      String state = getPullRequestState(prNumber);
+      return MERGED_STATE.equalsIgnoreCase(state);
+    }
+
+    public Category getCategoryFromPullRequest(String prNumber) throws Exception {
+      BufferedReader br =
+          runSubProcessAndGetOutputAsReader(
+              format(
+                  "gh pr view %s --repo %s/%s --jq \".labels[].name\" --json labels",
+                  prNumber, this.owner, this.repository));
+
+      String line;
+      while ((line = br.readLine()) != null) {
+        if (isValidCategory(line)) break;
+      }
+      if (line == null || line.isEmpty()) line = Category.MISCELLANEOUS.name();
```

Comment:
Why don't we return `Category.MISCELLANEOUS`here?

Reply:
Ah, it was bad. I'll revise it to return `Category.MISCELLANEOUS` directory. Thank you.

Reply:
Done. 0d498b8